main.js

```diff
@@ -24,35 +53,38 @@ let todoComponent = div(
         () => [todos]
     ),
     dynamic(
-        () => list(todos.map((todo) => p(todo))),
+        () => list(todos.map((todo) => div(link(todo, "/todo/" + todo + "/info?param1=test&hello=world")))),
         () => [todos]
     )
 );
 
-let rootDiv = div(
-    h1("Hoi!"),
-    h2("Hoi2!"),
+// setup router
+useRouter().on("/", mainPage);
 
-    span("Red text here :)").color("red"),
-
-    p("This is a paragraph.").on("dblclick", () => alert("Hello Universe")),
-
-    dynamic(
-        () => p(`Counter: ${counter} `, counter % 2 === 0 && "is Even"),
-        () => [counter]
-    ),
+// access router
+router()
+    .on("/todo", (params) => {
+        if (params["items"]) {
+            params["items"].split(",").forEach((item) => {
+                todo = item;
+                addTodo();
+            });
+        }
 
-    button("Click me!")
-        .on("click", function () {
-            console.log(this);
-            this.color("blue");
+        console.log(todos);
+        update();
+        return todoPage;
+    })
+    .on("/todo/:item/info", (params, vars) => {
+        let e = div(h1(vars["item"]));
 
-            counter++;
-        })
-        .color("#44DD44"),
+        for (const key in params) {
+            e.append(p(key + ": " + params[key]));
+        }
 
-    button("Reset").on("click", () => window.confirm("You sure?") && (counter = 0)),
-    todoComponent
-);
+        e.append(link("<- back", "/todo/"));
 
-document.body.append(rootDiv.HTML);
+        return e;
+    })
+    // redirect to active page
+    .sync();
```

Comment:
What does the `sync` method do? is it really necessary?

Reply:
`sync` is short for `redirect(window.location.pathname, true)`